test/test_jit.py

```diff
@@ -668,6 +669,24 @@ def f(x, y):
         y = torch.randn(4, 4, dtype=torch.float, device='cuda')
 
         ge = self.checkTrace(f, (x, y))
+        self.assertExpectedGraph(ge.graph_for(x,y))
+
+    @unittest.skipIf(IS_WINDOWS, "NYI: fuser support for Windows")
+    @unittest.skipIf(not RUN_CUDA, "fuser requires CUDA")
+    @skipIfRocm
+    def test_comparison_eq_ne(self):
+        def f(x, y):
+            mask = (x == 0).type_as(x)
+            z = x * mask + y
+            mask = (x != 0).type_as(x)
+            z = z * mask + y
+            return z
+
+        x = torch.randn(4, 4, dtype=torch.float, device='cuda')
+        y = torch.randn(4, 4, dtype=torch.float, device='cuda')
+
+        ge = self.checkTrace(f, (x, y))
+        self.assertExpectedGraph(ge.graph_for(x,y))
```

Comment:
Since the only thing you want to check here is that the whole graph was fused, it would be better to do this instead of adding more expects:
```py
def assertAllFused(self, graph):
  nodes = list(graph.nodes())
  self.assertEqual(len(nodes), 1)
  self.assertEqual(nodes[0].kind(), 'prim::FusionGroup')

def test_comparison(self):
   ...
   self.assertAllFused(ge.graph_for(x, y))
```

It would be even better if we could auto-generate those traces instead of writing them out manually. It's very easy - just construct closures of different operators.

Reply:
Arbitrary number of `int = prim::Constant` nodes gets added to list(graph.nodes()), in these particular cases len(nodes) is 3. I can't find a good way to filter them out other than grepping for something `aten::` or `Float(*`, which seems incredibly fragile.

Reply:
I'm confused about your comment. The expects you added have no constant nodes at the top level, so what I proposed should still be easy to do.

Reply:
Well those nodes aren't even used, so they could be easily eliminated with our DCE pass. I agree that internal documentation is definitely not our string side, but all other functions that are needed to implement this have been available for a few months at least.